internal/ingest/httpingest/httpingest.go

```diff
@@ -51,6 +52,24 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request, params api.In
 		namespace = *params.NamespaceInput
 	}
 
+	// Deduping in the API can't guarantee consistency due to the parallel ingestion of events.
+	// Currently the only consistent deduplication is provided by the ksqlDB Processor.
+	if h.Dedupe != nil {
+		isUnique, err := h.Dedupe.Set(r.Context(), event)
+		if err != nil {
+			logger.ErrorCtx(r.Context(), "unable to set dedupe", "error", err)
+
+			models.NewStatusProblem(r.Context(), err, http.StatusInternalServerError).Respond(w, r)
+			return
+		}
+		if !isUnique {
+			logger.DebugCtx(r.Context(), "event is not unique")
+			w.WriteHeader(http.StatusOK)
+			return
+		}
+		logger.DebugCtx(r.Context(), "event is unique")
+	}
```

Comment:
I'm thinking this should be a collector implementation in the ingest package and we should wrap the "real" collector as a sort of delegate or middleware. But the HTTP layer is definitely not where it should be.